sentry: cmd_end_transaction.go:497: error resolving intent at %s on end transaction [%s]: %s | roachpb.Span; roachpb.TransactionStatus; io error while pread offset <redacted> len <redacted> <redacted> cannot allocate memory

This issue was autofiled by Sentry. It represents a crash or reported error on a live cluster with telemetry enabled.

Sentry link: https://sentry.io/organizations/cockroach-labs/issues/988093075/?project=164528&referrer=webhooks_plugin

Panic message:

>cmd_end_transaction.go:497: error resolving intent at %s on end transaction [%s]: %s | roachpb.Span; roachpb.TransactionStatus; io error while pread offset <redacted> len <redacted> <redacted> cannot allocate memory


<details>
<summary>Stacktrace (expand for inline code snippets):</summary>

https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/batcheval/cmd_end_transaction.go#L314-L316 in pkg/storage/batcheval.evalEndTransaction
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/replica_command.go#L92-L94 in pkg/storage.evaluateCommand
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/replica.go#L6348-L6350 in pkg/storage.evaluateBatch
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/replica.go#L6089-L6091 in pkg/storage.(*Replica).evaluateWriteBatchWithLocalRetries
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/replica.go#L6027-L6029 in pkg/storage.(*Replica).evaluateWriteBatch
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/replica.go#L3405-L3407 in pkg/storage.(*Replica).evaluateProposal
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/replica.go#L3357-L3359 in pkg/storage.(*Replica).requestToProposal
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/replica.go#L3619-L3621 in pkg/storage.(*Replica).propose
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/replica.go#L3252-L3254 in pkg/storage.(*Replica).tryExecuteWriteBatch
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/replica.go#L3096-L3098 in pkg/storage.(*Replica).executeWriteBatch
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/replica.go#L2025-L2027 in pkg/storage.(*Replica).sendWithRangeID
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/replica.go#L1973-L1975 in pkg/storage.(*Replica).Send
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/store.go#L3183-L3185 in pkg/storage.(*Store).Send
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/stores.go#L184-L186 in pkg/storage.(*Stores).Send
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/server/node.go#L986-L988 in pkg/server.(*Node).batchInternal.func1
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/util/stop/stopper.go#L302-L304 in pkg/util/stop.(*Stopper).RunTaskWithErr
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/server/node.go#L973-L975 in pkg/server.(*Node).batchInternal
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/server/node.go#L1014-L1016 in pkg/server.(*Node).Batch
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/rpc/context.go#L430-L432 in pkg/rpc.internalClientAdapter.Batch
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/transport.go#L198-L200 in pkg/kv.(*grpcTransport).sendBatch
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/transport.go#L168-L170 in pkg/kv.(*grpcTransport).SendNext
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/transport.go#L168-L170 in pkg/kv.(*grpcTransport).SendNext
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/dist_sender.go#L1318-L1320 in pkg/kv.(*DistSender).sendToReplicas
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/dist_sender.go#L390-L392 in pkg/kv.(*DistSender).sendRPC
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/dist_sender.go#L468-L470 in pkg/kv.(*DistSender).sendSingleRange
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/dist_sender.go#L1094-L1096 in pkg/kv.(*DistSender).sendPartialBatch
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/dist_sender.go#L770-L772 in pkg/kv.(*DistSender).divideAndSendBatchToRanges
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/dist_sender.go#L682-L684 in pkg/kv.(*DistSender).Send
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/txn_coord_sender.go#L233-L235 in pkg/kv.(*txnLockGatekeeper).SendLocked
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/txn_interceptor_metrics.go#L57-L59 in pkg/kv.(*txnMetrics).SendLocked
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/txn_interceptor_span_refresher.go#L166-L168 in pkg/kv.(*txnSpanRefresher).sendLockedWithRefreshAttempts
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/txn_interceptor_span_refresher.go#L104-L106 in pkg/kv.(*txnSpanRefresher).SendLocked
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/txn_interceptor_pipeliner.go#L168-L170 in pkg/kv.(*txnPipeliner).SendLocked
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/txn_interceptor_intent_collector.go#L105-L107 in pkg/kv.(*txnIntentCollector).SendLocked
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/txn_interceptor_sequence_nums.go#L61-L63 in pkg/kv.(*txnSeqNumAllocator).SendLocked
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/txn_interceptor_heartbeat.go#L229-L231 in pkg/kv.(*txnHeartbeat).SendLocked
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/kv/txn_coord_sender.go#L647-L649 in pkg/kv.(*TxnCoordSender).Send
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/internal/client/db.go#L622-L624 in pkg/internal/client.(*DB).sendUsingSender
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/internal/client/txn.go#L796-L798 in pkg/internal/client.(*Txn).Send
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/internal/client/txn.go#L610-L612 in pkg/internal/client.(*Txn).rollback
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/internal/client/txn.go#L527-L529 in pkg/internal/client.(*Txn).CleanupOnError
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/internal/client/db.go#L588-L590 in pkg/internal/client.(*DB).Txn
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/replica_command.go#L350-L352 in pkg/storage.(*Replica).adminSplitWithDescriptor
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/split_queue.go#L170-L172 in pkg/storage.(*splitQueue).processAttempt
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/split_queue.go#L126-L128 in pkg/storage.(*splitQueue).process
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/queue.go#L768-L770 in pkg/storage.(*baseQueue).processReplica
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/storage/queue.go#L646-L648 in pkg/storage.(*baseQueue).processLoop.func1.2
https://github.com/cockroachdb/cockroach/blob/89ce71e9b733df3855b370aa20bb809db3d4362c/pkg/util/stop/stopper.go#L323-L325 in pkg/util/stop.(*Stopper).RunAsyncTask.func1
</details>

```
pkg/storage/batcheval/cmd_end_transaction.go in pkg/storage/batcheval.evalEndTransaction at line 315
pkg/storage/replica_command.go in pkg/storage.evaluateCommand at line 93
pkg/storage/replica.go in pkg/storage.evaluateBatch at line 6349
pkg/storage/replica.go in pkg/storage.(*Replica).evaluateWriteBatchWithLocalRetries at line 6090
pkg/storage/replica.go in pkg/storage.(*Replica).evaluateWriteBatch at line 6028
pkg/storage/replica.go in pkg/storage.(*Replica).evaluateProposal at line 3406
pkg/storage/replica.go in pkg/storage.(*Replica).requestToProposal at line 3358
pkg/storage/replica.go in pkg/storage.(*Replica).propose at line 3620
pkg/storage/replica.go in pkg/storage.(*Replica).tryExecuteWriteBatch at line 3253
pkg/storage/replica.go in pkg/storage.(*Replica).executeWriteBatch at line 3097
pkg/storage/replica.go in pkg/storage.(*Replica).sendWithRangeID at line 2026
pkg/storage/replica.go in pkg/storage.(*Replica).Send at line 1974
pkg/storage/store.go in pkg/storage.(*Store).Send at line 3184
pkg/storage/stores.go in pkg/storage.(*Stores).Send at line 185
pkg/server/node.go in pkg/server.(*Node).batchInternal.func1 at line 987
pkg/util/stop/stopper.go in pkg/util/stop.(*Stopper).RunTaskWithErr at line 303
pkg/server/node.go in pkg/server.(*Node).batchInternal at line 974
pkg/server/node.go in pkg/server.(*Node).Batch at line 1015
pkg/rpc/context.go in pkg/rpc.internalClientAdapter.Batch at line 431
pkg/kv/transport.go in pkg/kv.(*grpcTransport).sendBatch at line 199
pkg/kv/transport.go in pkg/kv.(*grpcTransport).SendNext at line 169
pkg/kv/transport.go in pkg/kv.(*grpcTransport).SendNext at line 169
pkg/kv/dist_sender.go in pkg/kv.(*DistSender).sendToReplicas at line 1319
pkg/kv/dist_sender.go in pkg/kv.(*DistSender).sendRPC at line 391
pkg/kv/dist_sender.go in pkg/kv.(*DistSender).sendSingleRange at line 469
pkg/kv/dist_sender.go in pkg/kv.(*DistSender).sendPartialBatch at line 1095
pkg/kv/dist_sender.go in pkg/kv.(*DistSender).divideAndSendBatchToRanges at line 771
pkg/kv/dist_sender.go in pkg/kv.(*DistSender).Send at line 683
pkg/kv/txn_coord_sender.go in pkg/kv.(*txnLockGatekeeper).SendLocked at line 234
pkg/kv/txn_interceptor_metrics.go in pkg/kv.(*txnMetrics).SendLocked at line 58
pkg/kv/txn_interceptor_span_refresher.go in pkg/kv.(*txnSpanRefresher).sendLockedWithRefreshAttempts at line 167
pkg/kv/txn_interceptor_span_refresher.go in pkg/kv.(*txnSpanRefresher).SendLocked at line 105
pkg/kv/txn_interceptor_pipeliner.go in pkg/kv.(*txnPipeliner).SendLocked at line 169
pkg/kv/txn_interceptor_intent_collector.go in pkg/kv.(*txnIntentCollector).SendLocked at line 106
pkg/kv/txn_interceptor_sequence_nums.go in pkg/kv.(*txnSeqNumAllocator).SendLocked at line 62
pkg/kv/txn_interceptor_heartbeat.go in pkg/kv.(*txnHeartbeat).SendLocked at line 230
pkg/kv/txn_coord_sender.go in pkg/kv.(*TxnCoordSender).Send at line 648
pkg/internal/client/db.go in pkg/internal/client.(*DB).sendUsingSender at line 623
pkg/internal/client/txn.go in pkg/internal/client.(*Txn).Send at line 797
pkg/internal/client/txn.go in pkg/internal/client.(*Txn).rollback at line 611
pkg/internal/client/txn.go in pkg/internal/client.(*Txn).CleanupOnError at line 528
pkg/internal/client/db.go in pkg/internal/client.(*DB).Txn at line 589
pkg/storage/replica_command.go in pkg/storage.(*Replica).adminSplitWithDescriptor at line 351
pkg/storage/split_queue.go in pkg/storage.(*splitQueue).processAttempt at line 171
pkg/storage/split_queue.go in pkg/storage.(*splitQueue).process at line 127
pkg/storage/queue.go in pkg/storage.(*baseQueue).processReplica at line 769
pkg/storage/queue.go in pkg/storage.(*baseQueue).processLoop.func1.2 at line 647
pkg/util/stop/stopper.go in pkg/util/stop.(*Stopper).RunAsyncTask.func1 at line 324
```

| Tag | Value |
|---|---|
| Cockroach Release | `v2.1.4` |
| Cockroach SHA: | 89ce71e9b733df3855b370aa20bb809db3d4362c |
| Platform | linux ppc64le |
| Distribution | CCL |
| Environment | development |
| Command | server |
| Go Version | `go1.10.4`|
| # of CPUs |160 |
| # of Goroutines |1431 |

--- a/pkg/storage/batcheval/cmd_end_transaction.go
+++ b/pkg/storage/batcheval/cmd_end_transaction.go
@@ -492,7 +492,7 @@ func resolveLocalIntents(
 			}
 			return nil
 		}(); err != nil {
-			log.Fatalf(ctx, "error resolving intent at %s on end transaction [%s]: %s", span, txn.Status, err)
+			return nil, errors.Wrapf(err, "resolving intent at %s on end transaction [%s]", span, txn.Status)
 		}
 	}
 	// If the poison arg is set, make sure to set the abort span entry.

